packages/sanity/src/form/inputs/files/ImageToolInput/imagetool/HotspotImage.tsx

```diff
@@ -31,154 +38,103 @@ export interface HotspotImageProps {
   onError?: (event: React.SyntheticEvent<HTMLImageElement, Event>) => void
   onLoad?: () => void
 }
-export interface HotspotImageState {
-  containerAspect: number | null
-}
-export class HotspotImage extends React.PureComponent<HotspotImageProps, HotspotImageState> {
-  static defaultProps = {
-    alignX: 'center',
-    alignY: 'center',
-    className: '',
-    crop: DEFAULT_CROP,
-    hotspot: DEFAULT_HOTSPOT,
-    aspectRatio: 'none',
-  }
-
-  state: HotspotImageState = {
-    containerAspect: null,
-  }
-
-  containerElement: HTMLDivElement | null = null
-  imageElement: HTMLImageElement | null = null
-
-  componentDidMount() {
-    const imageElement = this.imageElement
+
+export const HotspotImage = memo(function HotspotImage(props: HotspotImageProps) {
+  const {
+    alignX = 'center',
+    alignY = 'center',
+    alt,
+    aspectRatio = 'none',
+    className = '',
+    crop = DEFAULT_CROP,
+    hotspot = DEFAULT_HOTSPOT,
+    onError,
+    onLoad,
+    src,
+    srcAspectRatio,
+    srcSet,
+    style,
+  } = props
+  const [containerAspect, setContainerAspect] = useState<number | null>(null)
+  const containerElement = useRef<HTMLDivElement | null>(null)
+  const imageElement = useRef<HTMLImageElement | null>(null)
```

Comment:
Personally I prefer to use `imageElementRef` convention, to make it easier to see that it's a `Ref` further down in the component code.

Reply:
Fair!